### ISSUE TITLE
Refactor sensu bin stubs to act as wrappers for configuring environment

### ISSUE THREAD

**Issue by cwjohnston** [sensu/sensu-build#174]
With systemd, smf and other service management systems we can't depend on being able to conditionally set environment variables (e.g. setting PATH and GEM_PATH based on value of EMBEDDED_RUBY).

As a result, we should investigate consolidating any such logic in refactored sensu bin stubs that can set up the environment properly before starting the process, ensure valid configuration for sane restarts, etc. This should give us consistent behavior across platforms without having to maintain the same behavior across multiple service management systems.

Duplicates sensu/omnibus-sensu#8 - code to close this issue will be ported to that repo if completed here first.

Closes #108
Closes #172

**Comment by amdprophet** [comment]
Won't fix here but these changes have been integrated in https://github.com/sensu/sensu-omnibus.
